filedistribution/src/main/java/com/yahoo/vespa/filedistribution/status/FileDistributionStatusClient.java

```diff
@@ -0,0 +1,236 @@
+        URI statusUri = createStatusApiUri();
+        if (debug)
+            System.out.println("URI:" + statusUri);
+        try {
+            CloseableHttpResponse response = httpClient.execute(new HttpGet(statusUri));
+            String content = EntityUtils.toString(response.getEntity());
+            if (debug)
+                System.out.println("response:" + content);
+            if (response.getStatusLine().getStatusCode() == 200) {
+                return content;
+            } else {
+                throw new RuntimeException("Failed to get status for request " + statusUri + ": " +
+                                                   response.getStatusLine() + ": " + content);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    String parseAndGenerateOutput(String json) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode jsonNode;
+        try {
+            jsonNode = objectMapper.readTree(json);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        String status = jsonNode.get("status").asText();
+        switch (status) {
+            case statusUnknown:
+                return "File distribution status unknown: " + jsonNode.get("message").asText();
+            case statusInProgress:
+                return "File distribution in progress:\n" + inProgressOutput(jsonNode.get("hosts"));
+            case statusFinished:
+                return "File distribution finished";
+            default:
+                throw new RuntimeException("Unknown status " + status);
+        }
+    }
+
+    private URI createStatusApiUri() {
+        String path = String.format("/application/v2/tenant/%s/application/%s/environment/%s/region/%s/instance/%s/filedistributionstatus",
+                                    tenantName, applicationName, environment, region, instanceName);
+        try {
+            return new URIBuilder()
+                    .setScheme("http")
+                    .setHost("localhost")
+                    .setPort(19071)
+                    .setPath(path)
+                    .addParameter("timeout", String.valueOf(timeout))
+                    .build();
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String inProgressOutput(JsonNode hosts) {
+        ArrayList<String> statusPerHost = new ArrayList<>();
+        for (JsonNode host : hosts) {
+            StringBuilder sb = new StringBuilder();
+            String status = host.get("status").asText();
+            sb.append(host.get("hostname").asText()).append(": ").append(status);
+            if (status.equals(statusUnknown))
+                sb.append(" (").append(host.get("message").asText()).append(")");
+            else if (status.equals(statusInProgress)) {
```

Comment:
I prefer braces, especially the when other branches has braces.

Reply:
Yeah, agree, will fix in a forthcoming PR